numpyro/distributions/continuous.py

```diff
@@ -63,6 +63,7 @@ def __init__(self, concentration1, concentration0, validate_args=None):
         super(Beta, self).__init__(batch_shape=batch_shape, validate_args=validate_args)
 
     def sample(self, key, sample_shape=()):
+        assert key is not None
```

Comment:
Should we instead `assert isinstance(key, random.PRNGKey)` instead? This will also work for `dist.Normal(0., 1.).sample((10,))`, but if you think the error raised in that case is meaningful enough, this should be fine.

Reply:
Good point! There is a hidden method in JAX to check if an array is a PRNGKey (random.PRNGKey alone is a function, not class). I think we can copy it to util and use it here.

Reply:
@neerajprad I have changed those assertations to `assert is_prng_key(key)`. This looks better now. Thanks!

Reply:
Looks great!